### PR TITLE
Msw 100037354 update feed item loading

### PR DESCRIPTION
**Pull request by michaelwehrley**

### CODE REVIEW

File: app/assets/javascripts/social_networking/controllers/home-controller.js [hunk 1]
```diff
@@ -61,7 +62,7 @@
   };
 
   HomeCtrl.prototype.retrieveFeed = function () {
-    var responsePromise = this._sharedResource.get('/social_networking/shared_items/participant/'+this._currentParticipantId+'/page/'+this.page);
+    var responsePromise = this._sharedResource.get('/social_networking/' + this._resource + '/participant/' + this._participantId + '/page/' + this.page);
```
Comment (eschlange) [inline app/assets/javascripts/social_networking/controllers/home-controller.js]: Are there circumstances where the resource changes to something other than a shared item?

File: app/controllers/social_networking/concerns/profile_page/feed.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
+module SocialNetworking
```
Comment (eschlange) [inline app/controllers/social_networking/concerns/profile_page/feed.rb]: I like the feed logic concern separation. Much better code cohesion!

File: app/controllers/social_networking/concerns/show_feed.rb [hunk 3]
```diff
@@ -29,30 +33,11 @@ def action_items
...
-        live_participants = group_participants.to_a.delete_if do |participant|
-          participant.active_group.nil? ||
-          participant.active_group.id != @participant.active_group.id
-        end
-
+        return unless active_group
         Serializers::ProfileSerializer.from_collection(
-          Profile.where(participant_id: live_participants.map(&:id)))
+          Profile.where(
+            participant_id: active_group.active_participants.pluck(:id)))
```
Comment (eschlange) [inline app/controllers/social_networking/concerns/show_feed.rb]: Does pluck speed things up, or is it just used to remove complexity from returned objects?

**Comment by eschlange** [comment]
:moyai:

File: app/assets/javascripts/social_networking/controllers/home-controller.js (hunk 1 above)
Comment (michaelwehrley) [inline app/assets/javascripts/social_networking/controllers/home-controller.js]: Yep.  Shared items for the profile page use the resource `profiles_page`.

File: app/controllers/social_networking/concerns/profile_page/feed.rb (hunk 2 above)
Comment (michaelwehrley) [inline app/controllers/social_networking/concerns/profile_page/feed.rb]: Thanks!

File: app/controllers/social_networking/concerns/show_feed.rb (hunk 3 above)
Comment (michaelwehrley) [inline app/controllers/social_networking/concerns/show_feed.rb]: I guess I could have used `map`, but it creates an array of ids.

**Pull request merged**